source/agora/test/Base.d

```diff
@@ -243,13 +258,16 @@ public class TestAPIManager
 
     public void printLogs ()
     {
-        import std.stdio;
-        foreach (key, api; this.apis)
+        synchronized  // make sure logging output is not interleaved
```

Comment:
Shouldn't the function be marked `synchronized` directly ?

Reply:
Hmm.. I didn't know you can use it like that. I'll read up on it.

Reply:
Actually from what I can tell `synchronized` on a method means it will synchronize on the object itself. But there are multiple `TestAPIManager` instances, one per unittest. I want to synchronize globally.

Reply:
What does it synchronize on without an object ?

Reply:
As far as I remember from TDPL, a global mutex, and I think each synchronized{} block has its own mutex.

Reply:
https://dlang.org/spec/statement.html#SynchronizedStatement

> What mutex is used is determined by the Expression. If there is no Expression, then a global mutex is created, one per such synchronized statement. Different synchronized statements will have different global mutexes.